src/Npgsql/TypeMapping/TypeMapper.cs

```diff
@@ -5,20 +5,20 @@
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
 using System.Reflection;
-using System.Runtime.CompilerServices;
 using Microsoft.Extensions.Logging;
 using Npgsql.Internal;
 using Npgsql.Internal.TypeHandlers;
 using Npgsql.Internal.TypeHandling;
 using Npgsql.Internal.TypeMapping;
 using Npgsql.PostgresTypes;
-using Npgsql.Properties;
-using Npgsql.Util;
 using NpgsqlTypes;
 
 namespace Npgsql.TypeMapping;
 
-sealed class TypeMapper
+/// <summary>
+/// Type mapper used to map types to type handlers.
+/// </summary>
+public sealed class TypeMapper
```

Comment:
Why specifically does this need to be public? If we have to, we should at least move it into the Internal namespace (and then remove the changes from the PublicAPI file)

Reply:
It's public because RangeResolver has to somehow resolve the handler for the actual type (e.g. Range<int>). So it makes sense that it's going to consult `TypeMapper` for it.
But yeah, I'll move it to internal namespace, just left it there for a time being.

Reply:
But isn't RangeResolver part of Npgsql, and can access TypeMapper as internal?

Reply:
Well, yes but actually no 😋

The way it's designed right now is we have a resolver factory `TypeHandlerResolverFactory`, on which we call `Create` to actually get the `TypeHandlerResolver`. And we only call it while initializing `TypeMapper`. So we can't really pass `TypeMapper` any other way other than through `TypeHandlerResolverFactory.Create` (which is public). And as another argument, it's kinda more fair for anyone designing custom handlers to have the same abilities as we are.

Reply:
Obviously we can work around this by accessing `TypeMapper` of a connector, but I would rather expand the public API and not look for the ways to do some weird stuff.

Reply:
Understood.

I think the requirement to have the TypeMapper inside the resolver really is very specific to the new range resolver; we're not aware of any other resolver that needs that (inside or outside Npgsql), and it's only the case with this resolver because it's a bit special (handles arbitrary range types). I'm a bit wary of opening up more and more internal API surface here, though I can't really think of a way to avoid it, and I guess if it's in the Internal namespace it's OK.